src/QMCApp/QMCMain.cpp

```diff
@@ -551,56 +552,64 @@ bool QMCMain::runQMC(xmlNodePtr cur, bool reuse)
   std::string prev_config_file = last_driver ? last_driver->get_root_name() : "";
   bool append_run              = false;
 
-  if (!population_)
+  // Within this scope if you split myComm you must catch UniformMPIError for your split MPI comm scope.
+  try
   {
```

Comment:
Is it possible to narrow this try region? I feel better to have it round where the throw may occur. Including the heavy run is less favourable to me.

Reply:
I'm not sure I understand, do we have a compiler/platform combination that doesn't do 'zero cost' exceptions? There are a number of considerations about the scope at which you want to catch an exception, but I think it is more expensive to developers to catch them in more places than necessary. Ideally when we logically acquired a comm scope we would enter a scope like we do with a ParallelExecutor so the catch scope and the comm context are matched.  Since that is not the way we implement Communicate concurrency right now this looked to me to be as large as safe but inside of all comm->splits.

Reply:
I'm just concerned about UniformMPIError potentially being abused, if someone didn't diligently add proper catch and assumed this one is safe to do all the catch. There is no way to prevent this kind of abusing from high level and such abuse may cause the code to hang. That is why I'd like to be extremely cautious about using UniformMPIError for large region. It doesn't need to be as large as possible. If the catch is missed, the generic catch of runtime_error still works. Having it large instead is dangerous. In your case, protect process() is probably enough.